AdyenNetworking/APIClient/APIClient.swift

```diff
@@ -148,15 +152,22 @@ public final class APIClient: APIClientProtocol {
         completionHandler: @escaping CompletionHandler<R.ResponseType>
     ) where R: Request, R.ResponseType == DownloadResponse {
         do {
-            urlSession.downloadTask(with: try buildUrlRequest(from: request)) { [weak self] result in
+            urlSession.downloadTask(with: try buildUrlRequest(from: request)) { [weak self]
+                result in
+                assert(!Thread.isMainThread, "Should not be running on main thread!!")
```

Comment:
the deadlock happen also if the main thread is waiting for the current background thread which in turn will wait for the main thread which would be a deadlock, in general, using `sync` is discouraged.